app/src/lib/find-account.ts

```diff
@@ -94,10 +70,33 @@ export async function findAccountForRemoteURL(
 
   const repositoryIdentifier = parseRepositoryIdentifier(urlOrRepositoryAlias)
   if (repositoryIdentifier) {
-    const { owner, name } = repositoryIdentifier
-    const account = await findRepositoryAccount(allAccounts, owner, name)
-    if (account) {
-      return account
+    const { owner, name, hostname } = repositoryIdentifier
+
+    // Prefer an authenticated dot com account, then Enterprise accounts, and
+    // finally the unauthenticated dot com account.
+    const sortedAccounts = Array.from(allAccounts).sort((a1, a2) => {
+      if (a1.endpoint === getDotComAPIEndpoint()) {
+        return a1.token.length ? -1 : 1
+      } else if (a2.endpoint === getDotComAPIEndpoint()) {
+        return a2.token.length ? 1 : -1
+      } else {
```

Comment:
Just curious how we determine an enterprise account here? I'd expect to see a comparison made against `getEnterpriseAPIURL`.

Reply:
> While working with things in-memory like this we just have to assume that an account that doesn't match getDotComAPIEndpoint() and has a token is a GHE account.

That's good to know. Could you add that as a comment?

Reply:
8fba6c32b